proxy/grpcproxy/register_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcproxy
+
+import (
+	"testing"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/coreos/etcd/integration"
+	"github.com/coreos/etcd/pkg/testutil"
+	"golang.org/x/net/context"
+)
+
+func TestRegister(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	kvts := newKVProxyServer([]string{clus.Members[0].GRPCAddr()}, t)
+	defer kvts.close()
+
+	// create a client and try to get key from proxy.
+	cfg := clientv3.Config{
+		Endpoints:   []string{kvts.l.Addr().String()},
+		DialTimeout: 5 * time.Second,
+	}
+	cli, err := clientv3.New(cfg)
+	if err != nil {
+		t.Fatalf("err = %v, want nil", err)
+	}
+
+	testPrefix := "test-name"
+	donec := Register(cli, testPrefix, clus.Members[0].GRPCAddr())
+	gresp, gerr := cli.Get(context.Background(), testPrefix, clientv3.WithPrefix())
```

Comment:
I think this should be tested using the grpc resolver instead of fetching the keys directly; the expectation is it will work with clientv3/naming, the details of how it works isn't super important to the test